### DISCUSSION TITLE
How to change input label for Ory Elements?

### DISCUSSION THREAD

**Discussion by meotimdihia** [ory/elements#118] [Q&A] [answered]
Currently, the email input is shown as `ID`. And users often ask me what is their ID. 
How to change it:

![image](https://github.com/ory/kratos/assets/300961/a6d056eb-d30b-4b34-ab25-0615f57e16f8)

### Identity Schema

![image](https://github.com/ory/kratos/assets/300961/c5aaf39a-53de-4deb-8ce2-dbbdc70e7b72)

**Comment by meotimdihia** [comment]
Does anyone have the answer? It is strange to show email's label as ID.

**Comment by vinckr** [comment]
Hello @meotimdihia 

At the moment you have to build a custom UI to change these labels and other styling of the UI. You can look at this documentation for some pointers: https://www.ory.sh/docs/kratos/bring-your-own-ui/custom-ui-overview 

Making this possible in the hosted UI is on the roadmap, so this should be possible to do without custom UI soon.

**Reply by meotimdihia** [reply]
Thanks for the reply. I am using `@ory/elements`. I switched from custom UI to it.

**Reply by Benehiko** [reply]
You can rewrite the flow data before passing it along to Ory Elements. Just swap out the property `meta.label.text` value for something else and it will change in the UI. 

```json5
...
"ui": {
  "nodes": [
    {
      "type": "input",
      "group": "default",
      "attributes": {
        "name": "identifier",
        "type": "text",
        "value": "",
        "required": true,
        "disabled": false,
        "node_type": "input"
      },
      "messages": [],
      "meta": {
        "label": {
            "id": 1070004,
            "text": "ID",
            "type": "info"
        }
      }
    }
  ],
}
```

**Reply by meotimdihia** [reply]
@Benehiko thanks, it works for me.

**Comment by Benehiko** [comment, marked as answer]
Just to also keep everyone up to date with the current status of this. We have recently introduced translations into Ory Elements and Ory Kratos and we will be providing a way of adding custom translations or custom text to a supported language through the Ory Elements `IntlProvider`. See this WIP https://github.com/ory/elements/pull/158
